cloudevents/README.md

```diff
@@ -0,0 +1,143 @@
+# Knative CloudEvents SDK
+
+This library produces CloudEvents in version 0.1 compatible form. To learn more
+about CloudEvents, see the [Specification](https://github.com/cloudevents/spec).
+
+There are two roles the SDK fulfills: the [producer](#producer) and the
+[consumer](#consumer). The producer creates a cloud event in either
+[Binary](#binary) or [Structured](#structured) request format. The producer
+assembles and sends the event through an HTTP endpoint. The consumer will
+inspect the incoming HTTP request and select the correct decode format.
+
+This SDK should be wire-compatible with any other producer or consumer of the
+supported versions of CloudEvents.
+
+## Getting Started
+
+CloudEvents acts as the envelope in which to send a custom object. Create a
+CloudEvent type for the events you will be producing.
+
+Example CloudEvent Type: `dev.knative.cloudevent.example`
+
+Select a source to identify the originator of this CloudEvent. It should be a
+valid URI.
+
+Example CloudEvent Source: `https://github.com/knative/pkg#cloudevents-example`
+
+And finally, create a struct that will be the data inside the CloudEvent,
+example:
+
+```go
+type Example struct {
+    Sequence int    `json:"id"`
+    Message    string `json:"message"`
+}
+```
+
+### Producer
+
+The producer will create a new `Example` object, fill out the CloudEvent struct,
+and post the event via a [Binary](#binary) or [Structured](#structured) request
+format.
+
+```go
+
+package main
+
+import (
+    "github.com/knative/pkg/cloudevents"
+    "log"
+)
+
+type Example struct {
+    Sequence int    `json:"id"`
+    Message  string `json:"message"`
+}
+
+func main() {
+    c := cloudevents.NewClient(
+        "dev.knative.cloudevent.example",
+        "https://github.com/knative/pkg#cloudevents-example",
+        "http://localhost:8080",
+    )
```

Comment:
Can we leave this here and agree to break people later on?I suspect it might become more clear once there is a registry or something and things will have to change again.